### PR TITLE
Do not append asset hashes on Sphinx 7.1+

### LINKED ISSUE #136: Theme broken with Sphinx 7.1.0
Since the release of Sphinx 7.1.0 we've been seeing

```
Theme error:
An error happened in rendering the page index.
Reason: OSError(22, 'Invalid argument')
```

in many of our projects. We're using the "python-docs-theme". Here's an (abbreviated) log of a minimal reproducer: 

```
PS E:\temp> py -3 -m virtualenv Sphinx
created virtual environment CPython3.11.4.final.0-64 in 764ms
  creator CPython3Windows(dest=E:\temp\Sphinx, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=C:\Users\TNR1\AppData\Local\pypa\virtualenv)
    added seed packages: pip==23.0.1, setuptools==67.6.0, wheel==0.40.0
  activators BashActivator,BatchActivator,FishActivator,NushellActivator,PowerShellActivator,PythonActivator
PS E:\temp> .\Sphinx\Scripts\pip install sphinx python-docs-theme
Collecting sphinx
  Using cached sphinx-7.1.0-py3-none-any.whl (3.2 MB)
Collecting python-docs-theme
  Using cached python_docs_theme-2023.5-py3-none-any.whl (17 kB)

... blah blah blah ...

PS E:\temp> cd .\Sphinx\
PS E:\temp\Sphinx> .\Scripts\sphinx-quickstart.exe --sep --project foo --author foo --quiet

Finished: An initial directory structure has been created.

You should now populate your master file E:\temp\Sphinx\source\index.rst and create other documentation
source files. Use the Makefile to build the docs, like so:
   make builder
where "builder" is one of the supported builders, e.g. html, latex or linkcheck.

PS E:\temp\Sphinx> .\Scripts\sphinx-build.exe -D html_theme=python_docs_theme source build
Running Sphinx v7.1.0
building [mo]: targets for 0 po files that are out of date
writing output...
building [html]: targets for 1 source files that are out of date
updating environment: [new config] 1 added, 0 changed, 0 removed
reading sources... [100%] index
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
copying assets... copying static files... done
copying extra files... done
done
writing output... [100%] index
Theme error:
An error happened in rendering the page index.
Reason: OSError(22, 'Invalid argument')
PS E:\temp\Sphinx>
```

I first reported this [as Sphinx issue](https://github.com/sphinx-doc/sphinx/issues/11519). There I was pointed to a [very similar issue](https://github.com/sphinx-doc/sphinx/issues/11518) in which it was concluded that the theme code needs to be changed. I guess that a similar fix is required here...?

### PR DESCRIPTION
**Pull request by pradyunsg**
Closes #136

### CODE REVIEW

**Comment by pradyunsg** [comment]
Whoops, I accidentally used the wrong repo to send the branch to! Sorry about that!

**Review by hugovk — APPROVED** [review]
Like https://github.com/pradyunsg/furo/commit/a92dd0c2bfa398825e239171925a81bdc428e90a, thank you!

Great that Sphinx has included this!

File: python_docs_theme/__init__.py [hunk 1]
```diff
@@ -20,6 +21,11 @@ def _asset_hash(path: str) -> str:
 
 
 def _add_asset_hashes(static: List[str], add_digest_to: List[str]) -> None:
+    if sphinx.version_info >= (7, 1):
```
Comment (AA-Turner) [inline python_docs_theme/__init__.py:24]: There's already a check on line 41, could we modify that to read:

```
    (4,) >= sphinx.version_info > (7, 1)
```

Comment (pradyunsg) [reply python_docs_theme/__init__.py:24]: (FWIW, that conditional is always going to be False)

**Comment by pradyunsg** [comment]
(requesting a new review, since the implementation changed)

File: python_docs_theme/__init__.py (hunk 1 above)
Comment (AA-Turner) [reply python_docs_theme/__init__.py]: Sorry, I'm currently on a train--should've waited until I got back to my computer to make sure I got the check right, sorry!

**Review by hugovk — APPROVED** [review]

File: python_docs_theme/__init__.py (hunk 1 above)
Comment (pradyunsg) [reply python_docs_theme/__init__.py]: Don't be! The review is appreciated! :)

**Pull request merged**